### ISSUE TITLE
[no-explicit-any] overriding the error message

### ISSUE THREAD

**Issue by nishantwrp** [typescript-eslint/typescript-eslint#1906]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-explicit-any": ["error"]
  }
}
```

```TS
// let a: any;
```

**Expected Result**
Is there a way I can override the error message with my custom error message?

**Actual Result**
"Specify a different type"

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `X.Y.Z` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
nope. could you explain more about what you're looking to do? Why is the default message not sufficient?

**Comment by nishantwrp** [comment]
The default message doesn't actually explain in detail what steps should the dev take to fix the error. However, I think this is a reasonable functionality and should be present.

**Comment by bradzacher** [comment]
> I think this is a reasonable functionality and should be present

There are very few ESLint rules that implement functionality to override the error message.

---

In a broad sense, "specify a different type" explains what needs to be done - the `any` needs to be replaced with a safer type.

What would you like the message to say?
Please provide as much information as you can. An example would be great
Help me to help you.

**Comment by nishantwrp** [comment]
The message could be something like 
`Please specify a more strict type instead of any. You can go to this link for instructions on finding the exact type.`

**Comment by bradzacher** [comment]
I mean the rule name is `no-explicit-any`, which shows up alongside the error message in the 
cli:
![Screen Shot 2020-04-16 at 11 48 51](https://user-images.githubusercontent.com/7462525/79494698-49709580-7fd8-11ea-92be-d5f48ac1cf91.png)

and in the ide (vscode here):
![image](https://user-images.githubusercontent.com/7462525/79494547-0ca49e80-7fd8-11ea-8219-64e446cb93af.png)
In VSCode, the rule name here is even clickable, and it takes you to the [complete documentation for the rule](https://github.com/typescript-eslint/typescript-eslint/blob/v2.28.0/packages/eslint-plugin/docs/rules/no-explicit-any.md), which explains (with examples) valid and invalid code for the rule.

"specify a more strict type instead of any" isn't any clearer than "specify a different type" IMO.

---

Who is your target audience? What is their skill level such that an error on an explicit any isn't enough explanation to solve the lint?

**Comment by nishantwrp** [comment]
Sorry for the delay @bradzacher. Actually our audience have a lot of beginner coders and we would like to redirect them to our own custom wiki page related to the instructions on adding type definitions.

**Comment by bradzacher** [comment]
I would suggest creating an extension rule locally in your project so you can completely control the documentation URLs, and extend the messages to say whatever you want.

```ts
const { rules: { 'no-explicit-any': rule } } = require(`@typescript-eslint/eslint-plugin`);

module.exports = {
  ...rule,
  meta: {
    ...rule.meta,
    docs: {
      ...rule.meta.docs,
      url: "https://some-docs-you-own.com",
    },
    messages: {
      ...rule.meta.messages,
      unexpectedAny: "Whatever error message you want",
    },
  },
};
```

**Comment by nishantwrp** [comment]
@bradzacher Ok, thanks.
